Update option B: Include parts of Bootstrap

o-alquimista opened the pull request:
By following the current instructions I was unable to add new colors to the `$theme-colors` variable. To make it work, I had to import variables first.

This won't work (as in "it doesn't result in the creation of `bg-<color>` classes"):
```css
@import "../node_modules/bootstrap/scss/functions";

$my-theme: (
  "blackhole": #111111,
  "onion": #59316b
);

$theme-colors: map-merge($theme-colors, $my-theme);

@import "../node_modules/bootstrap/scss/variables";
@import "../node_modules/bootstrap/scss/mixins";
```

This will work:
```css
@import "../node_modules/bootstrap/scss/functions";
@import "../node_modules/bootstrap/scss/variables";

$my-theme: (
  "blackhole": #111111,
  "onion": #59316b
);

$theme-colors: map-merge($theme-colors, $my-theme);

@import "../node_modules/bootstrap/scss/mixins";
```

I'm not very experienced with Sass, though. Maybe I did something wrong. Please check if you can reproduce this.
I'm using Dart Sass 1.33.0.

o-alquimista committed:
Update option B: Include parts of Bootstrap

By following the current instructions I was unable to add new colors to the `$theme-colors` variable. To make it work, I had to import variables first.

This won't work (as in "it doesn't result in the creation of `bg-<color>` classes"):
```css
@import "../node_modules/bootstrap/scss/functions";

$my-theme: (
  "blackhole": #111111,
  "onion": #59316b
);

$theme-colors: map-merge($theme-colors, $my-theme);

@import "../node_modules/bootstrap/scss/variables";
@import "../node_modules/bootstrap/scss/mixins";
```

This will work:
```css
@import "../node_modules/bootstrap/scss/functions";
@import "../node_modules/bootstrap/scss/variables";

$my-theme: (
  "blackhole": #111111,
  "onion": #59316b
);

$theme-colors: map-merge($theme-colors, $my-theme);

@import "../node_modules/bootstrap/scss/mixins";
```

I'm not very experienced with Sass, though. Maybe I did something wrong. Please check if you can reproduce this.
I'm using Dart Sass 1.33.0.

ffoodd commented:
That's not the solution indeed. The docs are right but are incomplete: variables overrides have to come before the variable file, but maps modification should come after since they need to exist to be customizable.

ffoodd commented on site/content/docs/5.0/customize/sass.md:
```diff
@@ -52,13 +52,13 @@ In your `custom.scss`, you'll import Bootstrap's source Sass files. You have two
 // 3. Include remainder of required Bootstrap stylesheets
-@import "../node_modules/bootstrap/scss/variables";
 @import "../node_modules/bootstrap/scss/mixins";
 
```
This is probably where we'd need to an example of map modification.

o-alquimista replied:
Between line 61 and 62, right?

Maybe a simple comment there would be enough.

```
// 4. Include any map modifications here
```

What do you think?

Because I'm inexperienced with Sass, I believed `$theme-colors` was a variable, not a "map". A comment like that would make the reader think about it...

ffoodd replied:
Probably, indeed. And we could mention it in the next part, dedicated to map customization.

ffoodd commented on site/content/docs/5.0/customize/sass.md:
```diff
@@ -52,13 +52,13 @@ In your `custom.scss`, you'll import Bootstrap's source Sass files. You have two
 
-// 1. Include functions first (so you can manipulate colors, SVGs, calc, etc)
+// 1. Include functions and variables first (so you can manipulate colors, SVGs, calc, etc)
 @import "../node_modules/bootstrap/scss/functions";
+@import "../node_modules/bootstrap/scss/variables";
 
 // 2. Include any default variable overrides here
 
 // 3. Include remainder of required Bootstrap stylesheets
-@import "../node_modules/bootstrap/scss/variables";
```
Those changes aren't needed and would in fact reintroduce other issues.

o-alquimista committed:
Merge branch 'twbs:main' into patch-1

o-alquimista committed:
Update importing and maps and loops section

Added a comment to indicate where maps can be modified in the "Importing" section, and added required imports to the "add to map" section

mdo commented:
Docs in `main` should be good to go compared to this PR, so closing out.
